Highway.cpp

```diff
@@ -1,34 +1,55 @@
 #include "Highway.h"
+#include "Car.h"
+#include "HighwayPatrol.h"
+#include "Motorcycle.h"
+#include "SemiTruck.h"
+
 
 #include <cassert>
 
 void Highway::changeSpeed(int newSpeed)
 {
     speedLimit = newSpeed;
-    for( auto* vehicle : vehicles )
-    {
-        vehicle->setSpeed(speedLimit);
-    }
 }
 
 void Highway::addVehicleInternal(Vehicle* v)
 {
-    assert(false);
-
     /*
     depending on the derived type, call the member function that doesn't evade the cops. 
     */
+    if(auto* car = dynamic_cast<Car*>(v))
+    {
+        car->closeWindows();
+    }
+    else if(auto* chopper = dynamic_cast<Motorcycle*>(v))
+    {
+        chopper->setSpeed(105);
+    }
+    else if(auto* truck = dynamic_cast<SemiTruck*>(v))
+    {
+        truck->setSpeed(60);
+    }
```

Comment:
you're supposed to call `chopper->lanesplitAndRace()` and whatever you wrote in your `SemiTruck` class
- [ ] revise